packages/block-library/src/block/edit.native.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * External dependencies
+ */
+import { Platform, Text, TouchableWithoutFeedback, View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef, useState } from '@wordpress/element';
+import { useEntityBlockEditor } from '@wordpress/core-data';
+import { BottomSheet, Icon, Disabled } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { BlockEditorProvider, BlockList } from '@wordpress/block-editor';
+import { usePreferredColorSchemeStyle } from '@wordpress/compose';
+import { help } from '@wordpress/icons';
+import { requestUnsupportedBlockFallback } from '@wordpress/react-native-bridge';
+import { store as reusableBlocksStore } from '@wordpress/reusable-blocks';
+
+/**
+ * Internal dependencies
+ */
+import styles from './editor.scss';
+import EditTitle from './edit-title';
+
+export default function ReusableBlockEdit( {
+	attributes: { ref },
+	clientId,
+	isSelected,
+} ) {
+	const recordArgs = [ 'postType', 'wp_block', ref ];
+
+	const [ showHelp, setShowHelp ] = useState( false );
+	const [ sendFallbackMessage, setSendFallbackMessage ] = useState( false );
+	const timeoutId = useRef();
+	const infoTextStyle = usePreferredColorSchemeStyle(
+		styles.infoText,
+		styles.infoTextDark
+	);
+	const infoTitleStyle = usePreferredColorSchemeStyle(
+		styles.infoTitle,
+		styles.infoTitleDark
+	);
+	const infoSheetIconStyle = usePreferredColorSchemeStyle(
+		styles.infoSheetIcon,
+		styles.infoSheetIconDark
+	);
+	const actionButtonStyle = usePreferredColorSchemeStyle(
+		styles.actionButton,
+		styles.actionButtonDark
+	);
+
+	const { reusableBlock, hasResolved, isEditing, settings } = useSelect(
+		( select ) => {
+			return {
+				reusableBlock: select( 'core' ).getEditedEntityRecord(
+					...recordArgs
+				),
+				hasResolved: select( 'core' ).hasFinishedResolution(
+					'getEditedEntityRecord',
+					recordArgs
+				),
+				isSaving: select( 'core' ).isSavingEntityRecord(
+					...recordArgs
+				),
+				canUserUpdate: select( 'core' ).canUser(
+					'update',
+					'blocks',
+					ref
+				),
+				isEditing: select(
+					reusableBlocksStore
+				).__experimentalIsEditingReusableBlock( clientId ),
+				settings: select( 'core/block-editor' ).getSettings(),
+			};
+		},
+		[ ref, clientId ]
+	);
+
+	const { invalidateResolution } = useDispatch( 'core' );
+
+	const [ blocks, onInput, onChange ] = useEntityBlockEditor(
+		'postType',
+		'wp_block',
+		{ id: ref }
+	);
+
+	useEffect( () => {
+		return () => {
+			/**
+			 * Invalidate entity record upon unmount to keep the reusable block udpated
+			 * in case it's modified through UBE
+			 */
+			invalidateResolution( 'getEntityRecord', recordArgs );
+		};
+	}, [] );
```

Comment:
The reusable blocks for now can only be edited through UBE (unsupported block editor), however the updates are not reflected in the editor.

For this problem I came out with 2 solutions:
### Update the block's content with the changes made in UBE ❌
Unfortunately this solution didn't work because the content is updated directly within the UBE's web view so the changes are not propagated.

### Refresh the block's content fetching it from the API ✅
I noticed that by default entity record requests (`getEditedEntityRecord`) are cached so the only way to force a refresh is by invalidating it (`invalidateResolution`).

After a block is edited in UBE, it recreates the component so I found that the best place to execute the invalidation is when the component is unmounted. This way when the component is mounted again it will request the entity record again and hence refresh the block's content.